plugins/woocommerce/client/blocks/assets/js/base/components/cart-checkout/form/prepare-form-fields.ts

```diff
@@ -45,6 +45,10 @@ const getSupportedCoreLocaleProps = (
 		);
 	}
 
+	if ( localeField.optionalLabel !== undefined ) {
```

Comment:
`fields.optionalLabel` is set above if in the locale config:

there's a label, and there's no optional label.

The newly added code will override the default optional label if one is defined in the locale config